### ISSUE TITLE
Cannot resolve keyword 'p' into field.

### ISSUE THREAD

**Issue by StevenMedina** [alsoicode/django-admin-sortable#210]
Hi, I am trying to sort my objects but when I go to the url the P of 'page' as I apply the order button throws me this error, I am using version 2.1.14, could you help me with it please.

![fullscreen_2_26_19__9_14_am](https://user-images.githubusercontent.com/9811859/53419243-7d899100-39a7-11e9-9317-32e8f9885e78.png)

**Comment by alsoicode** [comment]
Hi. I'll need to see your Model and ModelAdmin definitions please.

**Comment by StevenMedina** [comment]
Okay, here they are
![fullscreen_2_26_19__9_37_am](https://user-images.githubusercontent.com/9811859/53420728-508aad80-39aa-11e9-8a35-0eb8b37b31a9.png)

![fullscreen_2_26_19__9_38_am](https://user-images.githubusercontent.com/9811859/53420756-626c5080-39aa-11e9-983c-5fc983cb806e.png)

![fullscreen_2_26_19__9_39_am](https://user-images.githubusercontent.com/9811859/53420793-757f2080-39aa-11e9-87cd-664f09ae9bb7.png)

**Comment by alsoicode** [comment]
Thanks. I'll need to make some time to add a sample model that matches your setup. In the meantime, you might drop back to version 2.1.11

**Comment by StevenMedina** [comment]
Perfect, I thank you and I remain pending.

**Comment by Alex12245** [comment]
The problem is that `IGNORED_PARAMS` does not contain `PAGE_VAR` variable. According to this 
problem, this condition   
`if k not in IGNORED_PARAMS:     
      filters[k] = v`  is always true for `p` param.

**Comment by alsoicode** [comment]
Ah, then we should be able to do: `if k not in IGNORED_PARAMS or k != PAGE_VAR`

**Comment by alsoicode** [comment]
@StevenMedina Can you please try the `fix-key-filtering` branch?

**Comment by StevenMedina** [comment]
@alsoicode Hi, sure, I check it and I'll tell you, thanks.

**Comment by StevenMedina** [comment]
@alsoicode  I have just done tests, and the part that was solved is that directly when I give to the next page it filters me without problem, but when I give it to organize it breaks again.

As I observe comes from this method which filters the queryset when I select the button to organize.

`def get_sort_view_queryset(self, request, sortable_by_expression):
        """
        Return a queryset, optionally filtered based on request and
        `sortable_by_expression` to be used in the sort view.
        """
        # get sort group index from querystring if present
        sort_filter_index = request.GET.get('sort_filter')
        filter_expression = request.GET.get('filter_expression')

        filters = {}

        if sort_filter_index:
            try:
                filters = self.model.sorting_filters[int(sort_filter_index)][1]
            except (IndexError, ValueError):
                pass

        if filter_expression:
            filters.update(
                dict([filter_expression.split('=')])
            )

        # Apply any sort filters to create a subset of sortable objects
        return self.get_queryset(request).filter(**filters)`


I will try to solve it and I will tell you if I succeed, thanks

**Comment by afk-mario** [comment]
Having the same issue here

``` py
class ImageProyecto(SortableMixin):
    order = models.PositiveIntegerField(
        default=0, editable=False, db_index=True)
    publish = models.BooleanField(default=True)
    show_on_home = models.BooleanField(default=False)
    width = models.PositiveIntegerField(default=0, editable=False)
    height = models.PositiveIntegerField(default=0, editable=False)
    note = models.CharField(max_length=140, blank=True, null=True)
    project = SortableForeignKey(
        'Proyecto', on_delete=models.CASCADE, related_name='images')
    image_type = models.CharField(
        max_length=2,
        choices=IMAGE_CHOICES,
        default='FR',
    )
    image = models.ImageField(
        'Imagen',
        upload_to=upload_to,
    )
    image_slider = ImageSpecField(
        source='image',
        processors=[
            ResizeToFill(
                width=IMAGE_SIZE_BIG_W, height=IMAGE_SIZE_BIG_H, upscale=False)
        ],
        format='PNG',
    )
    image_slider_small = ImageSpecField(
        source='image',
        processors=[
            ResizeToFill(
                width=IMAGE_SIZE_SMALL_W,
                height=IMAGE_SIZE_SMALL_H,
                upscale=False)
        ],
        format='PNG',
    )
    image_slider_webp = ImageSpecField(
        source='image',
        processors=[
            ResizeToFill(
                width=IMAGE_SIZE_BIG_W, height=IMAGE_SIZE_BIG_H, upscale=False)
        ],
        format='WEBP',
    )
    image_slider_small_webp = ImageSpecField(
        source='image',
        processors=[
            ResizeToFill(
                width=IMAGE_SIZE_SMALL_W,
                height=IMAGE_SIZE_SMALL_H,
                upscale=False)
        ],
        format='WEBP',
    )

    def save(self, *args, **kwargs):
        self.width = self.image.width
        self.height = self.image.height
        super(ImageProyecto, self).save(*args, **kwargs)

    def __str__(self):
        return self.image.url

    class Meta:
        ordering = ['order']
        verbose_name = 'Imagen de Proyecto'
        verbose_name_plural = 'Imagenes de proyecto'

```

**Comment by StevenMedina** [comment]
@alsoicode sorry for the delay, this method works perfectly, I no longer present the error
```python
def get_querystring_filters(self, request):
        filters = {}

        for k, v in request.GET.items():
            if k not in IGNORED_PARAMS and k != PAGE_VAR:
                filters[k] = v

        return filters
```

**Comment by alsoicode** [comment]
@StevenMedina I have released version 2.1.15 to PyPi with this fix for you.
